2024/XenSummit/Challenges_and_Status_of_Enabling_TrenchBoot_in_Xen_Hypervisor.md

```diff
@@ -0,0 +1,234 @@
+
+- **D-RTM Configuration Environment (DCE) Preamble** - responsible for
+  platform configuration and initiating **Dynamic Launch Event (DL Event)**
+- **D-RTM Configuration Environment (DCE)** - entered via **DL Event**,
+  performs DLME measurement and hands off the control to it
+- **Dynamically Launched Measured Environment (DLME)** - the ultimate result
+  of the dynamic launch, which effectively can be OS or bare metal software
+- D-RTM does not care about initial measurements since it use PCR[17-22]
+  which are locked until DL Event unlocks them in TPM locality 4
+
+---
+
+# TrenchBoot
+
+.center[.image-50[![](/img/tb_logo.svg)]]
+
+- [TrenchBoot Mailing List](https://groups.google.com/forum/#!forum/trenchboot-devel)
+- The `#trenchboot` channel on [OSFW Slack](https://slack.osfw.dev)
+  * Also bridged to Matrix as [#OSFW-Trenchboot:matrix.org](https://matrix.to/#/#OSFW-Trenchboot:matrix.org)
+- Twitter [@TrenchBoot](https://twitter.com/trenchboot)
+
+???
+
+TrenchBoot is a framework that allows individuals and projects to build
+security engines to perform launch integrity actions for their systems. The
+framework builds upon Boot Integrity Technologies (BITs) that establish one or
+more Roots of Trust (RoT) from which a degree of confidence that integrity
+actions were not subverted is derived.
+
+The first TrenchBoot D-RTM implementation has been made for GRUB+Linux
+
+---
+
+# TrenchBoot vs TrustedBoot
+
+.left-column50[
+
+### TrustedBoot
+
+- TrustedBoot (tboot) supports only Intel TXT
+- Is an exokernel and Xen (or any other kernel) has to be aware of its
+  presence
+]
+.right-column50[
+
+### TrenchBoot
+
+- Aims for unified approach supporting both AMD and Intel
+  processors
+- The goal is to implement a native support for D-RTM to let Xen have full
+  control without any exokernels
+]
+
+???
+
+Some may know about tboot - Intel TXT reference implementation
+
+With native support for DRTM, there are some additional steps to be performed
+before the CPUs may be initialized (wait for SIPI state). There is also a
+matter of handling power state transitions and event log.
+
+---
+
+# Usecase: Qubes OS Anti Evil Maid
+
+.center[.image-30[![](/img/qubes-logo-home.svg)]]
+
+- Qubes OS Anti Evil Maid (AEM) is a set of software packages and utilities to
+  aid against [Evil Maid attacks](https://theinvisiblethings.blogspot.com/2009/10/evil-maid-goes-after-truecrypt.html)
+- Leverages DRTM and TPM to seal secrets, which are used by the owner to
+  confirm whether the device has been tampered with or not
+- Initially only Intel TXT and TPM1.2 with tboot was supported using legacy
+  BIOS boot mode, now TrenchBoot replaces tboot, extending the support to
+  TPM2.0 and AMD SKINIT
+
+---
+
+# Demo: Qubes OS Anti Evil Maid
+
+.center[ <a href="https://asciinema.org/a/661348?cols=96&rows=24"
+target="_blank"> <img src="https://asciinema.org/a/661348.svg" width="400px"/>\
+</a> ]
```

Comment:
> There are 60+s where nothing happens, so we should cut it. There is no value for attendees in watching the blinking cursor

Yes, that's probably some bug we haven't covered, Xen is pretty much slow early after dynamic launch.

> it would be best to export it to mp4 and give Staszek for editing (add intro, outro, cut time when nothing happens), then upload to YT, so there is some additional marketing value from the effort of recording this video

Sure, but I would avoid showing intro/outro to comply with non-marketing guidelines for the presentation

> there is no information on what HBoM and SBoM this demo was created - it is very important for future reference to avoid statements "it worked at some point"

I hope my PRs will be merged and it will be tagged as v0.4.1, then I can add the components versions.

Reply:
> Yes, that's probably some bug we haven't covered, Xen is pretty much slow early after dynamic launch.

It is worth documenting that as a bug and provide as a reference.

> Sure, but I would avoid showing intro/outro to comply with non-marketing guidelines for the presentation

Do we have some written guidelines? I would like to realize what prevents videos with intro/outro. Does the guideline say that no brand can be mentioned or shown? If that is the case, we cannot show the company logo.

> I hope my PRs will be merged and it will be tagged as v0.4.1, then I can add the components versions.

If we explain what distributions can use the package, it is okay, but then we should also say what is inside the package (show release notes?) to explain what version of the components is important. HBoM is also may be important, as well as the version of firmware.

Reply:
> Do we have some written guidelines? I would like to realize what prevents videos with intro/outro. Does the guideline say that no brand can be mentioned or shown? If that is the case, we cannot show the company logo.

It was on the CFP site, but can't find that right now. it said to not promote products. SHowign a flashy demo with intro/outro is pushing it too much I think. There was no details what is actually allowed or not.

Reply:
Issue added: https://github.com/3mdeb/conferences/pull/31/commits/fe1f444b41f5daf3a9278f4d71082fffc51e7dca#diff-28488ef9a30ac840715c1e53410d8d69b64e939b4fd9fc4ea63380479fc4dc43R175

> If we explain what distributions can use the package, it is okay, but then we should also say what is inside the package (show release notes?) to explain what version of the components is important. HBoM is also may be important, as well as the version of firmware.

Small BOM added: https://github.com/3mdeb/conferences/pull/31/commits/fe1f444b41f5daf3a9278f4d71082fffc51e7dca#diff-28488ef9a30ac840715c1e53410d8d69b64e939b4fd9fc4ea63380479fc4dc43R170-R174

Still waiting for the PRs to be merged so I can create a new release on GH where the changes could be explained.


Currently we have packages only for QubesOS...

Reply:
> There are 60+s where nothing happens, so we should cut it. There is no value for attendees in watching the blinking cursor

I would skip it anyways during the presentation, but I have uploaded a new cut version.

Reply:
Great. Thank you.